src/components/Bridge/BridgeCard/index.tsx

```diff
@@ -1,111 +1,291 @@
+    estimatedAmount,
+    amountNet,
+    selectedRoute,
+    transactionLoaderStatus,
+    transactionError,
+    transactionStatus,
+  } = useDerivedBridgeInfo();
 
-  const BridgePrioritizationItems = [
-    {
-      label: t('bridge.bridgePrioritizations.recommended'),
-      value: 'recommended',
-    },
-    {
-      label: t('bridge.bridgePrioritizations.fast'),
-      value: 'fast',
-    },
-    {
-      label: t('bridge.bridgePrioritizations.normal'),
-      value: 'normal',
-    },
-  ];
+  const [inputCurrencyList, setInputCurrencyList] = useState<BridgeCurrency[]>([]);
+  const [outputCurrencyList, setOutputCurrencyList] = useState<BridgeCurrency[]>([]);
+  const [chainList, setChainList] = useState<Chain[] | undefined>(undefined);
+  const chainHook = useBridgeChains();
+  const [allBridgeCurrencies, setAllBridgeCurrencies] = useState<BridgeCurrency[]>([]);
+  const currencyHook = useBridgeCurrencies();
+  const sdkChainId = useChainId();
+  const [drawerType, setDrawerType] = useState(ChainField.FROM);
 
-  const Bridges = [
-    {
-      label: 'Thorchain',
-      value: 'thorchain',
-    },
-    {
-      label: 'DeFiChain',
-      value: 'defichain',
-    },
-  ];
-  const Exchanges = [
-    {
-      label: 'Pangolin',
-      value: 'pangolin',
-    },
-    {
-      label: 'Uniswap',
-      value: 'uniswap',
-    },
-    {
-      label: '1Inch',
-      value: '1inch',
+  const { library } = useLibrary();
+
+  const { sendTransaction } = useBridgeSwapActionHandlers();
+
+  const onSendTransaction = useCallback(() => {
+    selectedRoute?.bridgeType?.id && sendTransaction[selectedRoute?.bridgeType?.id](library, selectedRoute, account);
+  }, [selectedRoute]);
+
+  const onChangeTokenDrawerStatus = useCallback(() => {
+    setIsCurrencyDrawerOpen(!isCurrencyDrawerOpen);
+  }, [isCurrencyDrawerOpen]);
+
+  const onChangeChainDrawerStatus = useCallback(() => {
+    setIsChainDrawerOpen(!isChainDrawerOpen);
+  }, [isChainDrawerOpen]);
+
+  const {
+    onSwitchTokens,
+    onSwitchChains,
+    onCurrencySelection,
+    onChainSelection,
+    onUserInput,
+    onChangeRouteLoaderStatus,
+    onClearTransactionData,
+  } = useBridgeActionHandlers();
+
+  const maxAmountInput: CurrencyAmount | undefined = maxAmountSpend(sdkChainId, currencyBalances[CurrencyField.INPUT]);
+
+  const debouncedAmountValue = useDebounce(parsedAmount?.toExact(), 1500);
+
+  useEffect(() => {
+    if (currencyHook && activeBridges) {
+      let data: BridgeCurrency[] = [];
+      Object.entries(currencyHook).forEach(([key, value]) => {
+        if (activeBridges.some((bridge) => bridge.value === key)) {
+          data = data
+            ?.concat(value)
+            ?.filter(
+              (val, index, self) =>
+                index === self.findIndex((t) => t?.symbol === val?.symbol && t?.chainId === val?.chainId),
+            );
+        }
+      });
+      setAllBridgeCurrencies(data || []);
+    }
+  }, [currencyHook?.[LIFIBridge.id], activeBridges]);
```

Comment:
can we use `useMemo` instead of `useEffect` here. Also below useEffect that are setting `setInputCurrencyList`, `setOutputCurrencyList`, `setChainList` all can be converted into useMemo I think.  there are many instances in this PR, in other file with same kind of improvements please check those also.

Reply:
Yeah definitely! It seems I forgot to convert them. All done now.